packages/scaffold-config/src/detect.ts

```diff
@@ -113,8 +118,34 @@ export async function detectFramework (projectPath: string): Promise<DetectFrame
  * END
  */
 
-export function detectLanguage ({ projectRoot, pkgJson, isMigrating = false }: { projectRoot: string, pkgJson: PkgJson, isMigrating?: boolean }): 'js' | 'ts' {
+type DetectLanguageParams = {
+  projectRoot: string
+  customConfigFile?: string | null
+  pkgJson: PkgJson
+  isMigrating?: boolean
+}
+
+export function detectLanguage ({ projectRoot, customConfigFile, pkgJson, isMigrating = false }: DetectLanguageParams): 'js' | 'ts' {
   try {
+    if (customConfigFile) {
+      debug('Evaluating custom Cypress config file \'%s\'', customConfigFile)
+      if (/.ts$/.test(customConfigFile)) {
+        debug('Custom config file is Typescript - using TS')
+
+        return 'ts'
+      }
+
+      if (/.js$/.test(customConfigFile)) {
```

Comment:
We might need to widen this check, we also support `cjs` and `mjs` (and soon `mts`, I hope, some people asked for that).

Here's the default list of files we look for with the extensions we support: https://github.com/cypress-io/cypress/blob/develop/packages/data-context/src/data/ProjectLifecycleManager.ts#L37-L42

Reply:
@lmiller1990 Excellent point - pushed an update that expands the checks for both default and custom config file extensions to support `cjs`, `mjs`, and `mts`. I know `mts` isn't officially supported yet, but I believe it should be safe to include here since this is just trying to figure out what language the project is trying to use - failure to actually parse/support the config file will occur elsewhere. Please let me know if you disagree

Reply:
Sounds good. `mts` probably won't work yet, but adding this now will save us doing it later on when do we support `mts`.